softwareconfiguration/model/src/main/java/ai/libs/softwareconfiguration/model/ComponentInstance.java

```diff
@@ -1,329 +1,327 @@
+			ComponentInstance instanceChosenForRequiredInterface = current.getSatisfactionOfRequiredInterfaces().get(selection.getX());
+			if (!selection.getY().equals("*") && !instanceChosenForRequiredInterface.getComponent().getName().equals(selection.getY())) {
+				return false;
+			}
+			current = instanceChosenForRequiredInterface;
+		}
+		return true;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((this.component == null) ? 0 : this.component.hashCode());
+		result = prime * result + ((this.parameterValues == null) ? 0 : this.parameterValues.hashCode());
+		result = prime * result + ((this.satisfactionOfRequiredInterfaces == null) ? 0 : this.satisfactionOfRequiredInterfaces.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(final Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (this.getClass() != obj.getClass()) {
+			return false;
+		}
+		ComponentInstance other = (ComponentInstance) obj;
+		if (this.component == null) {
+			if (other.component != null) {
+				return false;
+			}
+		} else if (!this.component.equals(other.component)) {
+			return false;
+		}
+		if (this.parameterValues == null) {
+			if (other.parameterValues != null) {
+				return false;
+			}
+		} else if (!this.parameterValues.equals(other.parameterValues)) {
+			return false;
+		}
+		if (this.satisfactionOfRequiredInterfaces == null) {
+			if (other.satisfactionOfRequiredInterfaces != null) {
+				return false;
+			}
+		} else if (!this.satisfactionOfRequiredInterfaces.equals(other.satisfactionOfRequiredInterfaces)) {
+			return false;
+		}
+		return true;
+	}
+
+	public String toComponentNameString() {
+		StringBuilder sb = new StringBuilder();
+		sb.append(this.getComponent().getName());
+		if (!this.satisfactionOfRequiredInterfaces.isEmpty()) {
+			sb.append(this.satisfactionOfRequiredInterfaces.entrySet().stream().map(x -> x.getValue().toComponentNameString()).collect(Collectors.toList()).toString());
+		}
+		return sb.toString();
+	}
+
+	@JsonIgnore
+	@Override
+	public String toString() {
+		Map<String, Object> fields = new HashMap<>();
+		fields.put("component", this.component);
+		fields.put("parameterValues", this.parameterValues);
+		fields.put("satisfactionOfRequiredInterfaces", this.satisfactionOfRequiredInterfaces);
+		return ToJSONStringUtil.toJSONString(this.getClass().getSimpleName(), fields);
+	}
+
+	/**
+	 * Returns the description of a <code>ComponentInstance</code> as a pretty print with indentation.
+	 *
+	 * @return A string representing this object in JSON format.
+	 * @throws IOException
+	 *             An IOException is thrown if the object cannot be serialized to a String.
+	 */
+	@JsonIgnore
+	public String getPrettyPrint() throws IOException {
+		return new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).writeValueAsString(this);
+	}
+
+	public String getNestedComponentDescription() {
```

Comment:
Do we really want to have this method in a ComponentInstance or does this rather belong to a util class?